Source/AwsCommonRuntimeKit/mqtt/MqttRequestResponseClient.swift

```diff
@@ -0,0 +1,361 @@
+
+/// A streaming operation is automatically closed (and an MQTT unsubscribe triggered) when its
+// destructor is invoked.
+public class StreamingOperation {
+  fileprivate var rawValue: OpaquePointer?  // <aws_mqtt_rr_client_operation>?
+
+  public init() {
+    // TODO: INIT THE MEMBERS
+    self.rawValue = nil
+  }
+
+  /// Opens a streaming operation by making the appropriate MQTT subscription with the broker.
+  public func open() {
+    // TODO: open the stream
+  }
+
+  deinit {
+    // TODO: close the oepration
+  }
+}
+
+// We can't mutate this class after initialization. Swift can not verify the sendability due to the class is non-final,
+// so mark it unchecked Sendable
+/// Request-response client configuration options
+public class MqttRequestResponseClientOptions: CStructWithUserData, @unchecked Sendable {
+
+  /// Maximum number of subscriptions that the client will concurrently use for request-response operations.
+  public let maxRequestResponseSubscription: Int
+
+  /// Maximum number of subscriptions that the client will concurrently use for streaming operations.
+  public let maxStreamingSubscription: Int
+
+  /// Duration, in seconds, that a request-response operation will wait for completion before giving up.
+  public let operationTimeout: TimeInterval
+
+  public init(
+    maxRequestResponseSubscription: Int, maxStreamingSubscription: Int,
+    operationTimeout: TimeInterval
+  ) {
+    self.maxStreamingSubscription = maxStreamingSubscription
+    self.maxRequestResponseSubscription = maxRequestResponseSubscription
+    self.operationTimeout = operationTimeout
+  }
+
+  func validateConversionToNative() throws {
+    do {
+      _ = try self.operationTimeout.secondUInt32()
+    } catch {
+      throw CommonRunTimeError.crtError(
+        CRTError(
+          code: AWS_ERROR_INVALID_ARGUMENT.rawValue,
+          context: "Invalid operationTimeout value"))
+    }
+  }
+
+  typealias RawType = aws_mqtt_request_response_client_options
+  func withCStruct<Result>(
+    userData: UnsafeMutableRawPointer?, _ body: (aws_mqtt_request_response_client_options) -> Result
+  ) -> Result {
+    var options = aws_mqtt_request_response_client_options()
+    options.max_request_response_subscriptions = self.maxRequestResponseSubscription
+    options.max_streaming_subscriptions = self.maxStreamingSubscription
+    if let _operationTimeout: UInt32 = try? self.operationTimeout.secondUInt32() {
+      options.operation_timeout_seconds = _operationTimeout
+    }
+    options.terminated_callback = MqttRRClientTerminationCallback
+    options.user_data = userData
+    return body(options)
+  }
+}
+
+internal func MqttRRClientTerminationCallback(_ userData: UnsafeMutableRawPointer?) {
+  // Termination callback. This is triggered when the native client is terminated.
+  // It is safe to release the request response client at this point.
+  // `takeRetainedValue()` would release the client reference. ONLY DO IT AFTER YOU NEED RELEASE THE CLIENT
```

Comment:
Trivial: "`takeRetainedValue()` will release the client reference. IT SHOULD ONLY BE CALLED AFTER YOU RELEASE THE CLIENT"